### PR TITLE
F T32243 move procedure bundle to core

### PR DESCRIPTION
**Pull request by bakridemos**
**Ticket:** https://yaits.demos-deutschland.de/T32243 

Description: move the procedureBundle to core

Note : 
- this pr has to be merged with https://github.com/demos-europe/demosplan-addon/pull/54.
- the demosplan-addon dependency has to be updated.
- composer.lock has to be pushed too und then can this pr merged

Delete the checkbox if it doesn't apply/isn't necessary.

- [X] Tests updated/created
- [X] Link all relevant tickets
- [X] Move the tickets on the board accordingly

### CODE REVIEW

**Review by AlexMenschoid — CHANGES_REQUESTED** [review]
All changes look good and I tested the branch a bit with one project and it looked good. But I still finde some usage of the `DemosPlanProcedureBundle`:
- services.yml
- DemosPlanProcedureAPIController
- DemosPlanProcedureController
- Some test-related files

Please check and address those as well. Otherwise looks really good

**Comment by bakridemos** [comment]
> All changes look good and I tested the branch a bit with one project and it looked good. But I still finde some usage of the `DemosPlanProcedureBundle`:
> 
> * services.yml
> * DemosPlanProcedureAPIController
> * DemosPlanProcedureController
> * Some test-related files
> 
> Please check and address those as well. Otherwise looks really good

all concerns are handled, there are a few more changes related to the removing from the Datasheet logic which had to be done here too

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementGeoService.php [hunk 1]
```diff
@@ -602,9 +603,9 @@ protected function getDataFromPoints($points, $geoResults, $tempStatementId)
      * This is in StatementService, since it's only used there. It's not really about the procedure, it's actually
      * about the statement.
      */
-    private function isStatementOfProcedurePartOfWind(int $windNumber, string $procedureId): bool
+    private function isStatementOfProcedurePartOfWind(int $windNumber, string $procedureId, DatasheetService $datasheetService): bool
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/Statement/StatementGeoService.php:606]: You need to inject this as dependency or pass it on calling

**Review by mussbach — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementGeoService.php (hunk 1 above)
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Logic/Statement/StatementGeoService.php]: thx, done in https://github.com/demos-europe/demosplan-core/pull/1408/commits/430c2f983f76c58dd7d8e0942923d8631aa82c2b

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php [hunk 2]
```diff
@@ -804,7 +804,7 @@ public function listAction(
             $templateVars['orgaBranding'] = $orgaBranding;
         }
 
-        $datasheetVersion = $globalConfig->getDatasheetVersion($procedure);
+        $datasheetVersion = $datasheetService->getDatasheetVersion($procedure);
```
[the remark this answers is not in the document]
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php:807]: fixed in https://github.com/demos-europe/demosplan-core/pull/1408/commits/c301c08739265e82ba6d955ff8e16f9627770a30

File: demosplan/DemosPlanCoreBundle/Event/Procedure/GetDatasheetVersionEvent.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public function getProcedureId(): string
+    {
+        return $this->procedure;
+    }
+
+    public function setDatasheetVersion(int $datasheetVersion)
+    {
+        $this->datasheetVersion = $datasheetVersion;
```
[the remark this answers is not in the document]
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Event/Procedure/GetDatasheetVersionEvent.php:33]: fixed in https://github.com/demos-europe/demosplan-core/pull/1408/commits/c301c08739265e82ba6d955ff8e16f9627770a30

File: demosplan/DemosPlanCoreBundle/Event/Procedure/GetDatasheetVersionEvent.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public function getDatasheetVersion(): int
+    {
+        return $this->datasheetVersion;
```
[the remark this answers is not in the document]
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Event/Procedure/GetDatasheetVersionEvent.php:38]: fixed in https://github.com/demos-europe/demosplan-core/pull/1408/commits/c301c08739265e82ba6d955ff8e16f9627770a30

File: demosplan/DemosPlanCoreBundle/Services/Map/GetFeatureInfo.php [hunk 5]
```diff
@@ -142,7 +150,7 @@ public function getUrl2()
         $url2 = $this->url2;
         $globalConfig = $this->getGlobalConfig();
         $procedure = $this->getProcedureArray();
-        $datasheetVersion = $globalConfig->getDatasheetVersion($procedure['id']);
+        $datasheetVersion = $this->datasheetService->getDatasheetVersion($procedure['id']);
```
[the remark this answers is not in the document]
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Services/Map/GetFeatureInfo.php:153]: fixed in https://github.com/demos-europe/demosplan-core/pull/1408/commits/c301c08739265e82ba6d955ff8e16f9627770a30

File: demosplan/DemosPlanCoreBundle/Services/Map/GetFeatureInfo.php [hunk 6]
```diff
@@ -164,7 +172,7 @@
         $globalConfig = $this->getGlobalConfig();
         $url2Layer = $globalConfig->getMapGetFeatureInfoUrl2Layer();
         $procedure = $this->getProcedureArray();
-        $dataSheetVersion = $globalConfig->getDatasheetVersion($procedure['id']);
+        $dataSheetVersion = $this->datasheetService->getDatasheetVersion($procedure['id']);
```
[the remark this answers is not in the document]
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Services/Map/GetFeatureInfo.php:175]: fixed in https://github.com/demos-europe/demosplan-core/pull/1408/commits/c301c08739265e82ba6d955ff8e16f9627770a30

File: demosplan/DemosPlanCoreBundle/Services/Map/GetFeatureInfo.php [hunk 7]
```diff
@@ -186,7 +194,7 @@
         $versionString = '';
         $globalConfig = $this->getGlobalConfig();
         $procedure = $this->getProcedureArray();
-        $dataSheetVersion = $globalConfig->getDatasheetVersion($procedure['id']);
+        $dataSheetVersion = $this->datasheetService->getDatasheetVersion($procedure['id']);
```
[the remark this answers is not in the document]
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Services/Map/GetFeatureInfo.php:197]: fixed in https://github.com/demos-europe/demosplan-core/pull/1408/commits/c301c08739265e82ba6d955ff8e16f9627770a30

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementGeoService.php [hunk 8]
```diff
@@ -604,7 +612,7 @@ protected function getDataFromPoints($points, $geoResults, $tempStatementId)
      */
     private function isStatementOfProcedurePartOfWind(int $windNumber, string $procedureId): bool
     {
-        return $windNumber === $this->globalConfig->getDatasheetVersion($procedureId);
+        return $windNumber === $this->datasheetService->getDatasheetVersion($procedureId);
```
[the remark this answers is not in the document]
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Logic/Statement/StatementGeoService.php:615]: fixed in https://github.com/demos-europe/demosplan-core/pull/1408/commits/c301c08739265e82ba6d955ff8e16f9627770a30

File: demosplan/DemosPlanCoreBundle/Event/Procedure/GetDatasheetVersionEvent.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Event/Procedure/GetDatasheetVersionEvent.php:33]: @bakridemos no, this is not resolved. GetDatasheetVersionEvent::$datasheetVersion does not exist

**Review by mussbach — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Controller/Procedure/DemosPlanProcedureController.php [hunk 9]
```diff
@@ -1987,7 +1986,7 @@ public function publicDetailAction(
             }
         }
         // T16602 display html datasheets only in Procedures "wind" Version 1 and 2
-        $templateVars['htmlAvailable'] = \in_array($globalConfig->getDatasheetVersion($procedureId), [1, 2], true);
+        $templateVars['htmlAvailable'] = \in_array($datasheetService->getDatasheetVersion($procedureId), [1, 2], true);
```
[the remark this answers is not in the document]
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Controller/Procedure/DemosPlanProcedureController.php:1989]: Weird, this looks like a false positive

File: demosplan/DemosPlanCoreBundle/Event/Procedure/GetDatasheetVersionEvent.php [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    private string $procedureId;
+
+    public function __construct(string $procedureId)
+    {
+        $this->procedureId = $procedureId;
+    }
+
+    public function getProcedureId(): string
+    {
+        return $this->procedure;
```
[the remark this answers is not in the document]
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Event/Procedure/GetDatasheetVersionEvent.php]: fixed in https://github.com/demos-europe/demosplan-core/pull/1408/commits/29f98cf41e5140c0e5bdc02bb732374c5bacdb8a

File: demosplan/DemosPlanCoreBundle/Event/Procedure/GetDatasheetVersionEvent.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (bakridemos) [reply demosplan/DemosPlanCoreBundle/Event/Procedure/GetDatasheetVersionEvent.php:34]: jup sorry, is fixed here https://github.com/demos-europe/demosplan-core/commit/29f98cf41e5140c0e5bdc02bb732374c5bacdb8a

File: tests/backend/base/FunctionalTestCase.php [hunk 11]
```diff
@@ -441,9 +441,7 @@ private function hasValidDateFormat($dateString): bool
      */
     protected function getProcedurePhases()
     {
-        $path = DemosPlanPath::getRootPath('demosplan/DemosPlanProcedureBundle/Resources/config');
-
-        return Yaml::parseFile($path.'/procedurephases.yml');
+        return Yaml::parseFile(DemosPlanPath::getProjectPath('app/Resources/DemosPlanCoreBundle/config/procedurephases.yml'));
```
Comment (mussbach) [inline tests/backend/base/FunctionalTestCase.php:444]: Why did you move this to the project? Shouldn't this be `DemosPlanPath::getConfigPath('procedure/procedurephases.yml')`?

**Review by mussbach — CHANGES_REQUESTED** [review]

File: tests/backend/base/FunctionalTestCase.php (hunk 11 above)
Comment (bakridemos) [reply tests/backend/base/FunctionalTestCase.php]: you are right, this was a mistake, is fixed in https://github.com/demos-europe/demosplan-core/pull/1408/commits/18727d3cd76cf634764de38ebe685828eab2630d

**Review by mussbach — APPROVED** [review]
lgtm, code review, phpstan and manual testing, thx a lot!
